### DISCUSSION TITLE
I get an extra space at the end of a word for single character surrounds

### DISCUSSION THREAD

**Discussion by grimm26** [nvim-mini/mini.nvim#188] [Q&A] [answered]
`saw"` turns `word` into `"word "`
I don't know why I am getting that extra space at the end and would like not to :)
I am just using a black setup for mini.surround.

**Comment by echasnovski** [comment, marked as answer]
> `saw"` turns `word` into `"word "` I don't know why I am getting that extra space at the end and would like not to :) I am just using a black setup for mini.surround.

Here is a breakdown of `saw"`:
- `sa` initiates adding surrounding. Waits for user to input any kind of motion/textobject (on which surrounding will be added).
- `w` serves as that motion/textobject. It is a word motion (see `:h w`) which moves cursor at the start of next word. Which means that surrounding will be added at the region between current cursor position (included) and up until start of next word (excluded). This means including whitespace after current word.
- `"` indicates which surrounding you want to add (pair `"` in this case).

So to not include whitespace after current word use `iw` textobject (see `:h iw`). In this case it would be `saiw"`.

**Reply by grimm26** [reply]
Aha! I knew there was something I was missing. I checked cpoptions thinking something was off there.
